engine/src/main/java/org/terasology/rendering/dag/RenderingGraph.java

```diff
@@ -0,0 +1,140 @@
+        pipeline.add("backdrop", BackdropNode.class);
+        pipeline.add("skyBands", SkyBandsNode.class);
+        pipeline.add("objectsOpaque", ObjectsOpaqueNode.class);
+        pipeline.add("chunksOpaque", ChunksOpaqueNode.class);
+        pipeline.add("chunksAlphaReject", ChunksAlphaRejectNode.class);
+        pipeline.add("overlays", OverlaysNode.class);
+        pipeline.add("firstPersonView", FirstPersonViewNode.class);
+        pipeline.add("lightGeometry", LightGeometryNode.class);
+        pipeline.add("directionalLights", DirectionalLightsNode.class);
+        pipeline.add("chunksRefractiveReflective", ChunksRefractiveReflectiveNode.class);
+        pipeline.add("outline", OutlineNode.class);
+        pipeline.add("ambientOcclusionPasses", AmbientOcclusionPassesNode.class);
+        pipeline.add("prePostComposite", PrePostCompositeNode.class);
+        pipeline.add("simpleBlendMaterials", SimpleBlendMaterialsNode.class);
+        pipeline.add("lightShafts", LightShaftsNode.class);
+        pipeline.add("initialPostprocessing", InitialPostProcessingNode.class);
+        pipeline.add("downSampleSceneAndUpdateExposure", DownSampleSceneAndUpdateExposureNode.class); // TODO: a rename needed :)
+        pipeline.add("toneMapping", ToneMappedSceneNode.class);
+        pipeline.add("bloomPasses", BloomPassesNode.class);
+        pipeline.add("blurPasses", BlurPassesNode.class);
+        pipeline.add("finalPostProcessing", FinalPostProcessingNode.class);
+        pipeline.analyze();
+        return pipeline;
+    }
+
+    public void analyze() {
+        topologicalOrder = getTopologicalOrder();
+        stateManager.findStateChanges(topologicalOrder);
+        analyzed = true;
+    }
+
+    public void add(AbstractNode node) {
+        nodeMap.put(node.getIdentifier(), node);
+    }
+
+    public <T extends AbstractNode> void add(String identifier, Class<T> type) {
+        // Attempt constructor-based injection first
+        context.put(String.class, identifier);
+        T node = InjectionHelper.createWithConstructorInjection(type, context);
+        // Then fill @In fields
+        InjectionHelper.inject(node, context);
+        node.initialise();
+        this.add(type.cast(node));
+    }
+
+    public Node get(String identifier) {
+        return nodeMap.get(identifier);
+    }
+
+    public void process() throws NotAnalyzedException {
+        if (!analyzed) {
+            throw new NotAnalyzedException("RenderingPipeline must be analyzed before processing step.");
+        }
+
+        for (Node node : this.topologicalOrder) {
+            PerformanceMonitor.startActivity("rendering/" + node.getIdentifier());
+            stateManager.prepareFor(node);
+            node.process();
+            PerformanceMonitor.endActivity();
+        }
+    }
+
+    // TODO: getTopologicalOrder will be a complex task, therefore calling  it multiple times will be expensive
+    private Collection<Node> getTopologicalOrder() {
```

Comment:
1. I think you can narrow its output type to a _list_ of Nodes rather than a generic collection.
2. I'm tempted of calling it "getNodes()", but I'd be happy with getNodesInTopologicalOrder().